components/org.wso2.carbon.identity.api.server.organization.management/org.wso2.carbon.identity.api.server.organization.management.v1/src/main/resources/org.wso2.carbon.identity.organization.management.yaml

```diff
@@ -522,6 +522,251 @@ paths:
+        '400':
+          $ref: '#/components/responses/BadRequest'
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '403':
+          $ref: '#/components/responses/Forbidden'
+        '500':
+          $ref: '#/components/responses/ServerError'
+        '501':
+          $ref: '#/components/responses/NotImplemented'
+  /organizations/{organization-id}/discovery:
+    get:
+      tags:
+        - Organization Discovery
+      description:
+        This API facilitates the retrieval of discovery attributes for an organization. It currently provides the capability to retrieve these attributes only from the primary organization.<br>
+        <b>Permission required:</b> <br>
+        * /permission/admin/manage/identity/organizationmgt/view <br>
+        <b>Scope required:</b> <br>
+        * internal_organization_view
+      summary:
+        Get discovery attributes of the organization.
+      operationId: organizationDiscoveryGet
+      parameters:
+        - name: organization-id
+          in: path
+          description: ID of the organization whose discovery attributes are to be fetched.
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Successful response
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/OrganizationDiscoveryAttributes'
+        '400':
+          $ref: '#/components/responses/BadRequest'
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '403':
+          $ref: '#/components/responses/Forbidden'
+        '404':
+          $ref: '#/components/responses/NotFound'
+        '500':
+          $ref: '#/components/responses/ServerError'
+        '501':
+          $ref: '#/components/responses/NotImplemented'
+    delete:
+      tags:
+        - Organization Discovery
+      description:
+        This API serves the purpose of deleting discovery attributes of an organization, with the current restriction that only the primary organization has the capability to perform this action.<br>
+        <b>Permission required:</b> <br>
+        * /permission/admin/manage/identity/organizationmgt/update <br>
+        <b>Scope required:</b> <br>
+        * internal_organization_update
+      summary:
+        Delete discovery attributes of an organization.
+      parameters:
+        - name: organization-id
+          in: path
+          description: ID of the organization whose discovery attributes are to be deleted.
+          required: true
+          schema:
+            type: string
+      responses:
+        '204':
+          description: Successfully deleted
+        '400':
+          $ref: '#/components/responses/BadRequest'
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '403':
+          $ref: '#/components/responses/Forbidden'
+        '404':
+          $ref: '#/components/responses/NotFound'
```

Comment:
do we need 404 status code, due to idempotent law

Reply:
removed